Custom registries do not work correctly

_Originally posted by @crstnbr in https://github.com/bcbi/CompatHelper.jl/issues/103#issuecomment-552196094_

> Hm. I obtain an error when using the suggestion above ([CompatHelper.yml](https://github.com/crstnbr/TestPackage2.jl/blob/master/.github/workflows/CompatHelper.yml)):

> See the output here: https://github.com/crstnbr/TestPackage2.jl/commit/1d9350e3a0774841c3b2f1516430d358d5c3ab47/checks?check_suite_id=303953267

> Error:
```
Run julia -e 'using CompatHelper, Pkg; CompatHelper.main(; registries = Pkg.Types.RegistrySpec[ Pkg.RegistrySpec( name = "General", uuid = "23338594-aafe-5451-b93e-139f81909106", url = "https://github.com/JuliaRegistries/General.git"), Pkg.RegistrySpec( name = "TrebstRegistry", uuid = "df7bba20-026d-11ea-1ecb-03517e44f23d", url = "https://github.com/crstnbr/TrebstRegistry.git") ] )'
Cloning into 'REPO'...
Already on 'master'
Your branch is up to date with 'origin/master'.
Cloning into 'General'...
ERROR: MethodError: no method matching iterate(::Pkg.Types.VersionSpec)
Closest candidates are:
  iterate(!Matched::Core.SimpleVector) at essentials.jl:604
  iterate(!Matched::Core.SimpleVector, !Matched::Any) at essentials.jl:604
  iterate(!Matched::ExponentialBackOff) at error.jl:214
  ...
Stacktrace:
 [1] in(::Nothing, ::Pkg.Types.VersionSpec) at ./operators.jl:1055
 [2] #make_pr_for_new_version#13(::GitHub.OAuth2, ::Bool, ::Bool, ::CompatHelper.DefaultBranch, ::String, ::typeof(CompatHelper.make_pr_for_new_version), ::Function, ::GitHub.Repo, ::CompatHelper.Package, ::Dict{CompatHelper.Package,Union{Nothing, Pkg.Types.VersionSpec}}, ::Dict{CompatHelper.Package,Union{Nothing, String}}, ::Dict{CompatHelper.Package,Union{Nothing, VersionNumber}}, ::Array{GitHub.PullRequest,1}, ::Array{String,1}, ::CompatHelper.GitHubActions) at /home/runner/.julia/packages/CompatHelper/UyDZd/src/new_versions.jl:61
 [3] #make_pr_for_new_version at ./array.jl:0 [inlined]
 [4] #make_pr_for_new_version#12(::GitHub.OAuth2, ::Bool, ::Bool, ::CompatHelper.DefaultBranch, ::String, ::typeof(CompatHelper.make_pr_for_new_version), ::Function, ::GitHub.Repo, ::Dict{CompatHelper.Package,Union{Nothing, Pkg.Types.VersionSpec}}, ::Dict{CompatHelper.Package,Union{Nothing, String}}, ::Dict{CompatHelper.Package,Union{Nothing, VersionNumber}}, ::Set{CompatHelper.Package}, ::Array{GitHub.PullRequest,1}, ::Array{String,1}, ::CompatHelper.GitHubActions) at /home/runner/.julia/packages/CompatHelper/UyDZd/src/new_versions.jl:21
 [5] #make_pr_for_new_version at ./array.jl:0 [inlined]
 [6] #main#1(::Array{Pkg.Types.RegistrySpec,1}, ::Bool, ::Bool, ::CompatHelper.DefaultBranch, ::String, ::typeof(CompatHelper.main), ::Function, ::Base.EnvDict, ::CompatHelper.GitHubActions) at /home/runner/.julia/packages/CompatHelper/UyDZd/src/main.jl:38
 [7] #main at ./none:0 [inlined] (repeats 4 times)
 [8] top-level scope at none:1
##[error]Process completed with exit code 1.
```

> Did I do something wrong?

--- a/src/main.jl
+++ b/src/main.jl
@@ -26,7 +26,7 @@ function main(precommit_hook::Function = update_manifests,
                                                           auth = auth,
                                                           master_branch = master_branch)
     get_latest_version_from_registries!(dep_to_latest_version,
-                                        default_registries)
+                                        registries)
     _all_open_prs = get_all_pull_requests(repo, "open"; auth = auth)
     _nonforked_prs = exclude_pull_requests_from_forks(repo, _all_open_prs)
     my_username = get_my_username(ci_cfg; auth = auth, env = env)

--- a/src/new_versions.jl
+++ b/src/new_versions.jl
@@ -58,7 +58,7 @@ function make_pr_for_new_version(precommit_hook::Function,
     current_compat_entry = dep_to_current_compat_entry[dep]
     current_compat_entry_verbatim = dep_to_current_compat_entry_verbatim[dep]
     latest_version = dep_to_latest_version[dep]
-    if !isnothing(current_compat_entry) && latest_version in current_compat_entry
+    if !isnothing(latest_version) && !isnothing(current_compat_entry) && latest_version in current_compat_entry
         @info("latest_version in current_compat_entry",
               current_compat_entry,
               latest_version,

